control/settings/controllers/settings.home.controller.js

```diff
@@ -17,49 +19,135 @@
-            // });
 
-            /*Background image area ends*/
+            Settings.setAutoPlay = (e) => {
+                let value = e.target.checked;
+                if (value != Settings.data.content.autoPlay) {
+                    if (value === true && Settings.data.content.forceAutoPlay) {
+                        Settings.data.content.forceAutoPlay = false;
+                        Settings.data.content.transferAudioContentToPlayList = Settings.data.content.forceAutoPlay;
+                    }
+                    if (value === true) Settings.data.design.skipMediaPage = true;
+                    Settings.data.content.autoPlay = value;
+                    MediaCenter.save(Settings.data).then(() => {});
+                }
+            };
+
+            Settings.setAutoPlayDelay = (option) => {
+                if (option.value != Settings.data.content.autoPlayDelay.value) {
+                    Settings.data.content.autoPlayDelay = option;
+                    MediaCenter.save(Settings.data).then(() => {});
+                }
+            };
+
+            Settings.setGlobalPlaylist = (e) => {
+                let value = e.target.checked;
+                if (value != Settings.data.content.globalPlaylist) {
+                    Settings.data.content.globalPlaylist = value;
+                    MediaCenter.save(Settings.data).then(() => {});
+                }
+            };
+
+            Settings.setGlobalPlaylistNavButton = (e) => {
+                let value = e.target.checked;
+                if (value != Settings.data.content.showGlobalPlaylistNavButton) {
+                    Settings.data.content.showGlobalPlaylistNavButton = value;
+                    MediaCenter.save(Settings.data).then(() => {});
+                }
+            };
+
+            Settings.setGlobalPlaylistPlugin = (pluginInstance) => {
+                Settings.data.content.globalPlaylistPlugin = pluginInstance;
+                MediaCenter.save(Settings.data).then(() => {});
+            };
+
+            Settings.autoPlayDelayOptions = [
+                { label: "Off", value: 0 },
+                { label: "1s", value: 1 },
+                { label: "2s", value: 2 },
+                { label: "3s", value: 3 },
+                { label: "5s", value: 5 },
+            ];
+
+            Settings.showPluginsDialog = () => {
+                // MCM Playlist plugin unique Id
+                const globalPlaylistPluginTypeId = "5adb9753-58df-437c-96c6-efc6c3af9b4b";
+                buildfire.pluginInstance.showDialog({}, (err, instances) => {
+                    if (err) {
+                        return buildfire.dialog.toast({
+                            message: "Error occured",
+                            type: "danger",
+                        });
+                    }
+                    if (instances && instances.length > 0) {
+                        if (instances[0].pluginTypeId === globalPlaylistPluginTypeId) {
```

Comment:
I understand that we are checking what plugin was selected but what happened if I want to test it with QA instance? Will it fail? Since prod and QA instance doesn't have the same plugin type id. How do we test this? Am I missing something?